include/odp/api/spec/crypto.h

```diff
@@ -310,6 +310,19 @@ typedef struct odp_crypto_session_param_t {
 	 */
 	uint32_t auth_digest_len;
 
+	/** Authentication Additional Authenticated Data (AAD) length in bytes
+	 *
+	 *  This parameter is used when AAD length is constant for all
+	 *  operations (packets) of the session. When AAD length varies, this
+	 *  must be set to zero and operation level AAD length parameter is
+	 *  used instead. Both session and operation level parameters must not
+	 *  be used simultaneously.
+	 *
+	 *  Use odp_crypto_auth_capability() for supported AAD lengths.
+	 *  The default value is zero.
+	 */
+	uint32_t auth_aad_len;
+
```

Comment:
Why this restriction? Wouldn't it be cleaner to say this is simply a session-level default and if individual ops have a non-zero AAD they serve as overrides?